src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -221,8 +221,8 @@ protected function getAllowedAttributes($classOrObject, array $context, $attribu
         }
 
         $groups = false;
-        if (isset($context[static::GROUPS]) && \is_array($context[static::GROUPS])) {
-            $groups = $context[static::GROUPS];
+        if (isset($context[static::GROUPS])) {
+            $groups = (array) $context[static::GROUPS];
```

Comment:
shouldn't this check is_scalar||is_array instead of casting objects?

Reply:
done